src/extension.ts

```diff
@@ -221,4 +221,130 @@ export function activate(
+                    const fileDecorations: FileDecoration[] = []
+
+                    if (commitCoverage.status === 'fulfilled' && commitCoverage.value) {
+                        const { files: reportFiles } = commitCoverage.value.commit.report
+                        for (const file of nonDirectoryFiles) {
+                            const report = reportFiles[file.path]
+                            if (!report) {
+                                continue
+                            }
+
+                            const ratio = parseInt(report.t.c, 10)
+
+                            fileDecorations.push({
+                                uri: file.uri,
+                                after: {
+                                    contentText: `${ratio}%`,
+                                    color: 'gray',
+                                    hoverMessage: `the coverage ratio is ${ratio}%`,
+                                },
+                                meter: {
+                                    value: ratio,
+                                    hoverMessage: `the coverage ratio is ${ratio}%`,
+
+                                    min: 0,
+                                    low: 25,
+                                    high: 80,
+                                    max: 100,
+                                    optimum: 100,
+                                },
+                            })
+                        }
+                    }
+
+                    // Iterate over tree coverage results to construct directory decorations
+                    const directoryDecorations: FileDecoration[] = []
+
+                    for (const result of treeCoverages) {
+                        if (result.status === 'rejected') {
+                            continue
+                        }
+
+                        const { treeCoverage, directory } = result.value
+
+                        if (!treeCoverage) {
+                            continue
+                        }
+
+                        const ratio = parseInt(treeCoverage.commit.folder_totals.coverage, 0)
+
+                        directoryDecorations.push({
+                            uri: directory.uri,
+                            // We want to show 0%
+                            after: {
+                                contentText: `${ratio}%`,
+                                color: 'gray',
+                                hoverMessage: `the coverage ratio is ${ratio}%`,
+                            },
+
+                            meter: {
+                                value: ratio,
+                                hoverMessage: `the coverage ratio is ${ratio}%`,
+
+                                min: 0,
+                                low: 25,
+                                high: 80,
+                                max: 100,
+                                optimum: 100,
+                            },
+                        })
```

Comment:
the config between this and above is duplicated, could that be in a function so it's always the same?